server/lib/session/init.ts

```diff
@@ -1,8 +1,14 @@
 import { Context } from 'koa'
+import { MatchmakingType } from '../../../common/matchmaking'
 import { Permissions } from '../../../common/users/permissions'
 import { SelfUser } from '../../../common/users/user-info'
 
-export default function initSession(ctx: Context, user: SelfUser, permissions: Permissions) {
+export default function initSession(
+  ctx: Context,
+  user: SelfUser,
+  permissions: Permissions,
+  lastQueuedMatchmakingType: MatchmakingType = MatchmakingType.Match1v1,
```

Comment:
At this point I would pass an object for this instead (leave `ctx` out of it, can probably share typing with the session stuff in some way)